overlord/ifacestate/helpers.go

```diff
@@ -719,3 +726,41 @@ func ensureSystemSnapIsPresent(st *state.State) error {
 	_, err := snapstate.CoreInfo(st)
 	return err
 }
+
+func hotplugTaskSetAttrs(task *state.Task, deviceKey, ifaceName string) {
+	task.Set("device-key", deviceKey)
+	task.Set("interface", ifaceName)
+}
+
+func hotplugTaskGetAttrs(task *state.Task) (deviceKey, ifaceName string, err error) {
+	if err = task.Get("interface", &ifaceName); err != nil {
+		return "", "", fmt.Errorf("internal error: failed to get interface name: %s", err)
+	}
+	if err = task.Get("device-key", &deviceKey); err != nil {
+		return "", "", fmt.Errorf("internal error: failed to get device key: %s", err)
+	}
+	return deviceKey, ifaceName, err
+}
+
+type HotplugSlotDef struct {
+	Name             string                 `json:"name"`
+	Interface        string                 `json:"interface"`
+	StaticAttrs      map[string]interface{} `json:"static-attrs,omitempty"`
+	HotplugDeviceKey string                 `json:"device-key"`
+}
+
+func getHotplugSlots(st *state.State) (map[string]HotplugSlotDef, error) {
+	var slots map[string]HotplugSlotDef
```

Comment:
These should probably use pointers.. it looks like an identity type instead of a value type, and there's no reason to have all the data inside the map itself.